test/control_flow.coffee

```diff
@@ -1288,6 +1288,16 @@ test "#4871: `else if` no longer output together ", ->
    }
    '''
 
+test "#4889: `for` loop unexpected behavior", ->
+  n = 1
+  result = []
+  for i in [0..n]
+    result.push i
+    for j in [(i+1)..n] by 1
```

Comment:
Let’s either have both `for` loops say `by 1` or have neither do so.

If it makes a difference, in the other PR maybe we can copy this test with the other case (like one test with no `by`, and the other test with two `by 1`s).

Reply:
Without `by 1` the result will be `[ 0, 1, 1, 2, 1 ]`. I used the same code as the user in #4889 to show the final (expected) result. Adding `by 1 ` to the first loop will make no difference, but it's not actually the point of this PR (or #4889).    
The second loop range is `[1..1]` in the first iteration and `[2..1]` in the second, and this is the cause of different results if `by 1` is used or not.   
The purpose of this PR was to revert the compilation of `for - range - by` so the result would be the same as before #4853.

Reply:
I don’t think we need a PR that produces the result the user expects, if we think that that result is wrong. I thought the plan was to have two PRs:
- one so that `by 1` and a lack of `by` are always equivalent
- one that fixes the range output to be what you think it should be

Reply:
Oh, sorry, my bad.   
In that case, if you agree,  I suggest just one PR which will cover both issues, as changed range output is prerequisite for correct `by 1 ` and w/o `by` output.